cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java

```diff
@@ -350,12 +348,15 @@ public KafkaBrokerConfigurationBuilder withListeners(
         writer.println("listener.security.protocol.map=" + String.join(",", securityProtocol));
         writer.println("listeners=" + String.join(",", listeners));
 
-        // Advertised listeners are not allowed on KRaft nodes with controller only role
         if (!isKraftControllerOnly) {
-            writer.println("advertised.listeners=" + String.join(",", advertisedListeners));
+            // Join broker and controller listeners.
+            // The controller listeners is not empty, if it's a combined node, or broker in ZooKeeper mode or in migration state but not when full KRaft.
+            brokerAdvertisedListeners.addAll(controllerAdvertisedListeners);
+            writer.println("advertised.listeners=" + String.join(",", brokerAdvertisedListeners));
             writer.println("inter.broker.listener.name=" + REPLICATION_LISTENER_NAME);
         } else if (node.controller() && kafkaMetadataConfigState.isZooKeeperToPostMigration()) {
             // needed for KRaft controller only as well until post-migration because it needs to contact brokers
+            writer.println("advertised.listeners=" + String.join(",", controllerAdvertisedListeners));
```

Comment:
I know this will be used in Kafka 3.9.0. However, it is a forbidden option in the previous Kafka versions and it caused the Kafka node to stop. Unless that changed already in Kafka 3.8.0 to not have it exit when this option is set, you will need to gate it by the Kafka version.

(And assuming it needs the gating, we maybe should consider moving it into a separate PR as it will be needed for any kind of the Kafka 3.9.0 support)

Reply:
You are right, I've just realised the changes for removing the restriction of advertised.listeners configuration didn't get included in 3.8.0 (didn't make it before the code freeze). When 3.9.0 is released and supported with Strimzi, either way, this option would need to be gated by the Kafka version, as we would not be able to configure it for the  older versions (3.8.0 and 3.7.0) that are still supported with Strimzi right?

Reply:
I think we will drop 3.7 when 3.8 is added, not 100% sure yet given the strange release cycles. In any case 3.8.0 will be definitely still supported.